velox/functions/prestosql/tests/ArrayPositionTest.cpp

```diff
@@ -0,0 +1,441 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <optional>
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class ArrayPositionTest : public FunctionBaseTest {};
+
+TEST_F(ArrayPositionTest, integer) {
+  auto arrayVector = makeNullableArrayVector<int64_t>(
+      {{1, std::nullopt, 3, 4},
+       {3, 4, 5},
+       {},
+       {std::nullopt},
+       {5, 6, std::nullopt, 8, 9},
+       {7},
+       {3, 9, 8, 7}});
+
+  auto testPosition = [&](std::optional<int64_t> search,
+                          const std::vector<std::optional<int64_t>>& expected) {
+    auto result = evaluate<SimpleVector<int64_t>>(
+        "array_position(c0, c1)",
+        makeRowVector({
+            arrayVector,
+            makeConstant(search, arrayVector->size()),
+        }));
+
+    assertEqualVectors(makeNullableFlatVector<int64_t>(expected), result);
+  };
+
+  testPosition(3, {3, 1, 0, 0, 0, 0, 1});
+  testPosition(4, {4, 2, 0, 0, 0, 0, 0});
+  testPosition(5, {0, 3, 0, 0, 1, 0, 0});
+  testPosition(7, {0, 0, 0, 0, 0, 1, 4});
+  testPosition(
+      std::nullopt,
+      {std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt,
+       std::nullopt});
+}
+
+TEST_F(ArrayPositionTest, integerDictionaryEncoding) {
+  auto arrayVector = makeNullableArrayVector<int64_t>(
+      {{1, std::nullopt, 3, 4},
+       {3, 4, 5},
+       {},
+       {std::nullopt},
+       {5, 6, std::nullopt, 8, 9},
+       {7},
+       {3, 9, 8, 7}});
+  auto newSize = arrayVector->size() * 2;
+  auto indices = makeIndices(newSize, [](auto row) { return row / 2; });
+  auto dictArrayVector = wrapInDictionary(indices, newSize, arrayVector);
+
+  auto testPosition = [&](std::optional<int64_t> search,
```

Comment:
nit: looks like testPosition is common and can be templatiezed, cant we refactor so it takes the arrays also as input and does the validation ? so you dont need to create a new fn just for dicttype, arraytype etc..

Reply:
Moved the repetitive code into templates. It looks much better now. Thank you for this suggestion!